aten/src/ATen/native/TensorTransformations.cpp

```diff
@@ -19,7 +19,7 @@ Tensor flip_cpu(const Tensor& self, IntList dims) {
 
   auto indices = std::vector<at::Tensor>(flip_dims_size);
   for (int64_t i = 0; i < flip_dims_size; i++) {
-    indices[i] = at::arange(self.size(i) - 1, -1, -1, self.type().toScalarType(at::kLong));
+    indices[i] = at::arange(self.size(flip_dims_v[i]) - 1, -1, -1, self.type().toScalarType(at::kLong));
```

Comment:
The reason the tests passed earlier was because of the fact that the flip function was being tested on square tensors. This meant that `self.size(i)` was same for all `i` causing the failure to not be caught.

Reply:
This looks good to me, thanks!

Reply:
Thanks for catching the bug! :)